### PR TITLE
Extend YOLO9-grade multi-GPU training support to the rest of the library

### PR DESCRIPTION
**Pull request by EHxuban11**
## What

Brings every trainable family up to the DDP quality bar set for YOLO9 in issue #484. Follow-up to #488/#531/#551.

- **Auto-spawn for classify + restore**: `@ddp_aware()` added to resnet, convnext, efficientnetv2, mobilenetv4, and nafnet `train()`, so `model.train(device="0,1")` works from a plain script (previously torchrun-only for these five).
- **`sync_bn=True` defaults for the remaining BatchNorm-heavy pure CNNs**: yolox (yolo7 now inherits), yolonas det+pose, rtmdet, picodet, fomo. Same rationale as YOLO9Config: per-rank BN stats on a small shard degrade the converged model. Strictly a no-op outside DDP, so single-GPU runs and the open #566 convergence work are unaffected.
- **Global loss normalizers** (exact single-GPU-equivalent DDP gradients, like yolo9's `_global_cls_norm` and DETR `num_boxes`):
  - yolox and yolo7 `num_fg`
  - rtmdet `avg_factor`
  - yolonas `assigned_scores_sum`, det and pose
  - picodet VFL `avg_factor` + box/DFL `weight_sum`; the loss docstring claimed this was already wired, now it is
  - All reductions go through the existing `all_reduce_avg_scalar` (clamp-before-divide), so single-GPU numerics are unchanged.
- **picodet DDP graph consistency**: normalizer collectives moved before the no-positive early return (deadlock risk otherwise), and the all-background path keeps the regression tower in the autograd graph so DDP static_graph sees a consistent used-parameter set.
- **Downstream clamp fix (review round 1)**: picodet `VarifocalLoss` and rtmdet `QualityFocalLoss`/`GIoULoss` re-clamped the already-sanitized `avg_factor` to >= 1, under-scaling sparse and all-background multi-GPU batches by up to 1/world_size. The helpers now honor the caller's sanitized denominator.
- **Cleanup**: removed dead `broadcast_ema_buffers`; `tools/ddp_torchrun_smoke.py` (its one remaining caller) rewritten for the current design: EMA is updated on every rank after DDP-synced steps, so the smoke check verifies cross-rank EMA consistency with no broadcast (params always, buffers too under SyncBN on CUDA).
- **Tests**: yolox 2-rank gradient-parity vs single-GPU; picodet and rtmdet 2-rank parity for positive-sparse and all-background batches (the all-background variants fail at ratio ~0.5 if a downstream clamp returns, verified by temporarily reintroducing one); sync_bn config guard for all BN-heavy families; ddp_aware wiring test for the five new families.

Not touched: DETR families (already all-reduce `num_boxes`; backbone BN under small batch without SyncBN matches their upstream recipes), fomo loss (mean-reduced CE, contract-compliant), inference-only families (no training to distribute).

## How was this tested?

- `tests/unit/test_ddp_loss_parity.py` (incl. 5 new parity tests), `test_ddp_syncbn.py`, `test_ddp_distributed.py`: all pass locally (Windows, CPU/Gloo spawn)
- Family regression sweep (yolox/yolo7/yolonas/rtmdet/picodet/fomo/nafnet/classifiers/config): 515 passed
- `tools/ddp_torchrun_smoke.py` run 2-rank on CPU/Gloo: sampler, ema_consistency, yolo9 all PASS

## Code provenance

All code in this PR is original, written for this PR. No code was ported, adapted, or derived from any third-party repository, in any form.

- The changes extend in-repository mechanisms that already exist on dev: the `all_reduce_avg_scalar` reduction, yolo9's `_global_cls_norm` pattern, the `ddp_aware` decorator, and the `sync_bn` config flag (all from issue #484 work, PRs #488/#531/#551).
- Upstream projects are mentioned in comments only as behavioral references, to document that the DDP normalizer semantics match what each family's original training recipe does (mmdetection's `reduce_mean` on the RTMDet cls avg_factor, Apache-2.0; SuperGradients' all_reduce of `assigned_scores_sum`, Apache-2.0; PP-PicoDet's nranks averaging, Apache-2.0). No source code from those repositories was read, copied, or adapted for this PR; the statements describe publicly documented training behavior. No GPL, AGPL, LGPL, non-commercial, proprietary, or unknown-license material was used.

### CODE REVIEW

File: libreyolo/models/picodet/loss.py [hunk 1]
```diff
@@ -519,29 +521,43 @@ def forward(
...
-        # VFL across the whole batch (single call), normalised by total positives.
-        avg_factor = max(num_pos_total, 1)
+        # VFL across the whole batch (single call), normalised by the global
+        # (DDP-reduced) positive count, mirroring upstream PP-PicoDet's
+        # nranks averaging (issue #484). Identical to ``max(num_pos_total, 1)``
+        # outside DDP. Collectives run before the no-positive early return so
+        # every rank reaches them.
+        avg_factor = all_reduce_avg_scalar(
+            num_pos_total, device=bbox_flat.device
+        )
```
[the remark this answers is not in the document]
Comment (EHxuban11) [reply libreyolo/models/picodet/loss.py:536]: Fixed in 498d9bea: VarifocalLoss now divides by the caller's sanitized denominator directly (no re-clamp). The new 2-rank parity test `test_external_loss_ddp_gradient_matches_single_gpu_all_background[picodet]` pins this; it fails at ratio ~0.5 if the clamp comes back (verified by temporarily reintroducing it).

File: libreyolo/models/rtmdet/loss.py [hunk 2]
```diff
@@ -507,7 +509,12 @@ def forward(
 
         bg_class_ind = self.num_classes
         pos_inds = ((labels >= 0) & (labels < bg_class_ind)).nonzero().squeeze(1)
-        avg_factor = max(float(assign_metrics.sum().item()), 1.0)
+        # Global (DDP-reduced) soft positive mass, mirroring upstream mmdet's
+        # ``reduce_mean`` on the cls avg_factor: dividing by the global factor
+        # keeps DDP's gradient averaging equivalent to single-GPU training on
+        # the same global batch (issue #484). Identical to the previous
+        # ``max(sum, 1)`` outside DDP.
+        avg_factor = all_reduce_avg_scalar(assign_metrics.sum())
```
[the remark this answers is not in the document]
Comment (EHxuban11) [reply libreyolo/models/rtmdet/loss.py:524]: Fixed in 498d9bea: QualityFocalLoss and GIoULoss now divide by the caller's sanitized denominator directly (no re-clamp). Pinned by the new 2-rank parity test `test_external_loss_ddp_gradient_matches_single_gpu_all_background[rtmdet]`.
